src/simulation/simulation_state.jl

```diff
@@ -112,24 +88,46 @@ function _initialize_model_states!(
                             length = value_counts,
                         ),
                     ),
+                    params[key][2],
```

Comment:
The `2` make this hard to read because I don't know what this Tuple is. Can it be a NamedTuple instead? Then you would use a field name here.